git_wrapper/remote.py

```diff
@@ -65,24 +73,38 @@ def fetch(self, remote="origin"):
             raise exceptions.ReferenceNotFoundException(msg)
 
         try:
-            remote.fetch()
+            if prune_tags and not prune:
+                # log warning if prune_tags is True but prune is False
+                self.logger.info("prune_tags was ignored because prune is False")
+                remote.fetch()
+            else:
+                remote.fetch(prune=prune, prune_tags=prune_tags)
         except git.GitCommandError as ex:
             msg = (f"Could not fetch remote {remote.name} ({remote.url}). "
                    f"Error: {ex}")
             raise exceptions.RemoteException(msg) from ex
 
-    def fetch_all(self):
+    def fetch_all(self, prune=False, prune_tags=False):
         """Refresh all the repo's remotes.
 
            All the remotes will be fetched even if one fails ; in this case a
            single exception containing the list of failed remotes is returned.
+
+           Optionally, specify prune to be True to allow
+           for the removal of local branches that don't
+           exist on the remotes. If you also wish to remove
+           local tags that don't exist on the remotes, specify
+           prune_tags to be true.
```

Comment:
<nit> True is capitalized elsewhere

Reply:
Can you make the s/true/True/ change here? Consistency is good. Otherwise I am ready to approve, nice work!